maven-core/src/main/java/org/apache/maven/internal/aether/DefaultRepositorySystemSessionFactory.java

```diff
@@ -261,23 +261,24 @@ else if ( request.isUpdateSnapshots() )
         }
         session.setAuthenticationSelector( authSelector );
 
-        String transport = request.getUserProperties().getProperty( MAVEN_TRANSPORT_KEY, MAVEN_TRANSPORT_WAGON );
-        if ( MAVEN_TRANSPORT_RESOLVER.equals( transport ) )
+        String transport = request.getUserProperties()
+                .getProperty( MAVEN_RESOLVER_TRANSPORT_KEY, MAVEN_RESOLVER_TRANSPORT_WAGON );
+        if ( MAVEN_RESOLVER_TRANSPORT_NATIVE.equals( transport ) )
         {
             // Make sure (whatever extra priority is set) that resolver native is selected
-            configProps.put( RESOLVER_FILE_TRANSPORTER_PRIORITY_KEY, RESOLVER_MAX_PRIORITY );
-            configProps.put( RESOLVER_HTTP_TRANSPORTER_PRIORITY_KEY, RESOLVER_MAX_PRIORITY );
+            configProps.put( NATIVE_FILE_TRANSPORTER_PRIORITY_KEY, RESOLVER_MAX_PRIORITY );
+            configProps.put( NATIVE_HTTP_TRANSPORTER_PRIORITY_KEY, RESOLVER_MAX_PRIORITY );
         }
-        else if ( MAVEN_TRANSPORT_WAGON.equals( transport ) )
+        else if ( MAVEN_RESOLVER_TRANSPORT_WAGON.equals( transport ) )
         {
             // Make sure (whatever extra priority is set) that wagon is selected
-            configProps.put( WAGON_TRANSPORTER_KEY_PRIORITY_KEY, RESOLVER_MAX_PRIORITY );
+            configProps.put( WAGON_TRANSPORTER_PRIORITY_KEY, RESOLVER_MAX_PRIORITY );
         }
-        else if ( !MAVEN_TRANSPORT_AUTO.equals( transport ) )
+        else if ( !MAVEN_RESOLVER_TRANSPORT_AUTO.equals( transport ) )
         {
             throw new IllegalArgumentException( "Unknown maven.transport=" + transport
```

Comment:
Either do `Unknown " + KEY + "=" + value` or `Uknown Maven Resolver transport 'value'".
Remember the pattern we used in Resolver for named lock factories.

Reply:
fixed